test/mp/test/foo.h

```diff
@@ -48,7 +48,7 @@ class FooImplementation
     void initThreadMap() {}
     int callback(FooCallback& callback, int arg) { return callback.call(arg); }
     int callbackUnique(std::unique_ptr<FooCallback> callback, int arg) { return callback->call(arg); }
-    int callbackShared(std::shared_ptr<FooCallback> callback, int arg) { return callback->call(arg); }
+    int callbackShared(std::shared_ptr<FooCallback> callback, int arg) { return callback->call(arg); } // NOLINT(performance-unnecessary-value-param)
```

Comment:
In commit "clang-tidy: Fix `performance-unnecessary-value-param` check" (0498ed86bc6d1a6bdbabf98f2e39468dc5c90823)

Note: I think it does make sense to suppress this lint warning than try to avoid it. Could potentially avoid it by changing this argument to `const shared_ptr` but there's currently no IPC serialization code to handle `const shared_ptr` only `shared_ptr`, so it does't compile. Also I don't think it would make sense to add const shared_ptr serialization code because most cases where you would pass `const shared<ptr>` it probably makes more sense to pass a plain const reference.